lib/cocoapods-core/specification/json.rb

```diff
@@ -49,25 +54,32 @@ def self.from_json(json)
 
     # Configures a new specification from the given hash.
     #
-    # @param  [Hash] the hash which contains the information of the
+    # @param  [Hash] hash the hash which contains the information of the
     #         specification.
     #
+    # @param  [Specification] parent the parent of the specification unless the
+    #         specification is a root.
+    #
     # @return [Specification] the specification
     #
     def self.from_hash(hash, parent = nil)
       spec = Spec.new(parent)
       attributes_hash = hash.dup
       subspecs = attributes_hash.delete('subspecs')
+      testspecs = attributes_hash.delete('testspecs')
       spec.attributes_hash = attributes_hash
       spec.test_specification = !attributes_hash['test_type'].nil?
-      if subspecs
-        spec.subspecs = subspecs.map do |s_hash|
-          Specification.from_hash(s_hash, spec)
-        end
-      end
+      spec.subspecs.concat(subspecs_from_hash(spec, subspecs)) if subspecs
```

Comment:
I'd just `return [] if subspecs.nil?` in `subspecs_from_hash` to avoid repeating the logic